### DISCUSSION TITLE
issues installing firedrake in macOS M3

### DISCUSSION THREAD

**Discussion by reyesvalgui** [firedrakeproject/firedrake#3722] [Firedrake support] [answered]
I have been trying to install firedrake on my Mac (OS 14.5), but unfortunately I cannot get the install to complete. I've been trying defining new environmental variables, installing pets manually but I did not have success
 I received in terminal  the output

Successfully cloned repository loopy
Checking out branch main
Successfully checked out branch main
Updating submodules.
Successfully updated submodules.
Installing petsc/
Depending on your platform, PETSc may take an hour or more to build!
Traceback (most recent call last):
  File "/Users/guillermoreyesvalencia/firedrake-install", line 1816, in <module>
    install("petsc/")
  File "/Users/guillermoreyesvalencia/firedrake-install", line 1043, in install
    build_and_install_petsc()
  File "/Users/guillermoreyesvalencia/firedrake-install", line 1155, in build_and_install_petsc
    check_call([python, "./configure", "PETSC_DIR={}".format(petsc_dir), "PETSC_ARCH={}".format(petsc_arch)] + petsc_options)
  File "/Users/guillermoreyesvalencia/firedrake-install", line 672, in check_call
    log.debug(subprocess.check_output(arguments, stderr=subprocess.STDOUT, env=os.environ).decode())
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/subprocess.py", line 466, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/Users/guillermoreyesvalencia/firedrake/bin/python', './configure', 'PETSC_DIR=/Users/guillermoreyesvalencia/firedrake/src/petsc', 'PETSC_ARCH=default', '--download-superlu_dist', '--download-hwloc-configure-arguments=--disable-opencl', '--download-suitesparse', '--with-debugging=0', '--download-ptscotch', '--download-bison', '--with-shared-libraries=1', '--download-mpich-configure-arguments=--disable-opencl', '--download-pastix', '--download-mumps', '--download-hwloc', '--LDFLAGS=-Wl,-ld_classic', '--with-fortran-bindings=0', "--download-openblas-make-options='USE_THREAD=0 USE_LOCKING=1 USE_OPENMP=0'", '--download-pnetcdf', '--download-hdf5', '--with-x=0', '--download-mpich', '--with-zlib', '--with-c2html=0', '--download-netcdf', '--CFLAGS=-Wno-implicit-function-declaration', '--download-scalapack', '--download-openblas', '--download-metis', '--download-hypre']' returned non-zero exit status 1.

**Comment by rckirby** [comment, marked as answer]
Installing with anaconda is not well-supported.  Disable it in your shell and try with homebrew?  On Aug 3, 2024, at 4:32 PM, reyesvalgui ***@***.***> wrote:﻿
I have been trying to install firedrake on my Mac (OS 14.5), but unfortunately I cannot get the install to complete. I've been trying defining new environmental variables, installing pets manually but I did not have success
I received in terminal  the output
Successfully cloned repository loopy
Checking out branch main
Successfully checked out branch main
Updating submodules.
Successfully updated submodules.
Installing petsc/
Depending on your platform, PETSc may take an hour or more to build!
Traceback (most recent call last):
File "/Users/guillermoreyesvalencia/firedrake-install", line 1816, in 
install("petsc/")
File "/Users/guillermoreyesvalencia/firedrake-install", line 1043, in install
build_and_install_petsc()
File "/Users/guillermoreyesvalencia/firedrake-install", line 1155, in build_and_install_petsc
check_call([python, "./configure", "PETSC_DIR={}".format(petsc_dir), "PETSC_ARCH={}".format(petsc_arch)] + petsc_options)
File "/Users/guillermoreyesvalencia/firedrake-install", line 672, in check_call
log.debug(subprocess.check_output(arguments, stderr=subprocess.STDOUT, env=os.environ).decode())
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/opt/anaconda3/lib/python3.11/subprocess.py", line 466, in check_output
return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/opt/anaconda3/lib/python3.11/subprocess.py", line 571, in run
raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/Users/guillermoreyesvalencia/firedrake/bin/python', './configure', 'PETSC_DIR=/Users/guillermoreyesvalencia/firedrake/src/petsc', 'PETSC_ARCH=default', '--download-superlu_dist', '--download-hwloc-configure-arguments=--disable-opencl', '--download-suitesparse', '--with-debugging=0', '--download-ptscotch', '--download-bison', '--with-shared-libraries=1', '--download-mpich-configure-arguments=--disable-opencl', '--download-pastix', '--download-mumps', '--download-hwloc', '--LDFLAGS=-Wl,-ld_classic', '--with-fortran-bindings=0', "--download-openblas-make-options='USE_THREAD=0 USE_LOCKING=1 USE_OPENMP=0'", '--download-pnetcdf', '--download-hdf5', '--with-x=0', '--download-mpich', '--with-zlib', '--with-c2html=0', '--download-netcdf', '--CFLAGS=-Wno-implicit-function-declaration', '--download-scalapack', '--download-openblas', '--download-metis', '--download-hypre']' returned non-zero exit status 1.

**Comment by reyesvalgui** [comment]
Thank you very much, it seems it worked well
